packages/labs/virtualizer/src/support/method-interception.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type Function = (...args: any[]) => any;
+/* eslint-enable @typescript-eslint/no-explicit-any */
+
+/**
+ * An MethodInterceptor function is defined as accepting a function it is intercepting
+ * or "wrapping" calls to.  It has the same signature as the original function, except
+ * that it has an additional first argument that is the "original" or wrapped
+ * function that may be called or not depending on the interceptor's logic.  The
+ * approach is analagous to calling super() in a subclass's method.
+ */
+export type MethodInterceptor<T extends Function> = (
+  originalFunction: T | undefined,
+  ...args: Parameters<T>
+) => ReturnType<T>;
```

Comment:
nice!

Reply:
I don't even want to tell you how long it took to resign myself to using an explicit `any` in the function type.  It was a wild ride through all the resources.  I had some really ugly expanded type definitions through that exploration.  Usually a sign one is on the wrong track...  :-p